apps/external_ledger_db/lib/external_ledger_db/wallet.ex

```diff
@@ -0,0 +1,174 @@
+# Copyright 2017-2019 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule ExternalLedgerDB.Wallet do
+  @moduledoc """
+  Ecto Schema representing external ledger wallets. A wallet is made up of
+  a unique address and the ID associated with it.
+  """
+  use Ecto.Schema
+  import Ecto.{Changeset, Query}
+  alias ExternalLedgerDB.Repo
+
+  @behaviour Utils.Ledgers.WalletSchema
+
+  @primary_key {:uuid, Ecto.UUID, autogenerate: true}
+  @timestamps_opts [type: :naive_datetime_usec]
+
+  @hot "hot"
+  @cold "cold"
+
+  def hot, do: @hot
+  def cold, do: @cold
+
+  @ethereum "ethereum"
+  @omg_network "omg_network"
+
+  def ethereum, do: @ethereum
+  def omg_network, do: @omg_network
+
+  schema "wallet" do
+    field(:address, :string)
+    field(:adapter, :string)
+    field(:primary, :boolean, default: false)
+    field(:type, :string)
+    field(:public_key, :string)
+    field(:encrypted_private_key, ExternalLedgerDB.Encrypted.String)
+    field(:metadata, :map, default: %{})
+    field(:encrypted_metadata, ExternalLedgerDB.Encrypted.Map, default: %{})
+
+    timestamps()
+  end
+
+  @doc """
+  Validate the wallet attributes.
+  """
+  @spec changeset(%__MODULE__{}, map()) :: Ecto.Changeset.t()
+  def changeset(%__MODULE__{} = wallet, attrs) do
+    wallet
+    |> cast(attrs, [
+      :address,
+      :adapter,
+      :primary,
```

Comment:
Should we have a specific changeset for the primary hot wallet? Since it won't happen often. An alternative would be to store it in the settings instead of inside the DB. Thoughts?

Reply:
Hmm I like the idea of having it as a settings instead. 👍

Reply:
Alright, so it'll be like the master account, let's do it 👍